charts/influxdb2/templates/job-setup-admin.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    app: "{{ template "influxdb.fullname" . }}"
+    chart: "{{ template "influxdb.chart" . }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+  name: {{ template "influxdb.fullname" . }}-create-admin-user
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-delete-policy": "before-hook-creation"
+spec:
+  activeDeadlineSeconds: 30
+  template:
+    metadata:
+      labels:
+        app: {{ template "influxdb.fullname" . }}
+        release: "{{ .Release.Name }}"
+    spec:
+      containers:
+      - name: {{ template "influxdb.fullname" . }}-create-admin-user
+        image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
+        env:
+          - name: INFLUXDB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "influxdb.fullname" . }}-auth
+                key: admin-password
+          - name: INFLUXDB_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "influxdb.fullname" . }}-auth
+                key: admin-token
+        command:
+          - bash
+        args:
+          - -c
+          - |
+            influx setup -f \
+            --host http://{{ template "influxdb.fullname" . }}:{{ .Values.service.port }} \
+            -o {{ .Values.adminUser.organization }} \
+            -b {{ .Values.adminUser.bucket }} \
+            -u {{ .Values.adminUser.user }} \
+            -p ${INFLUXDB_PASSWORD} \
+            -t $INFLUXDB_TOKEN
```

Comment:
I think that `-t $INFLUXDB_TOKEN` may not be needed; I glanced quickly at the code and I believe -t reads that environment variable.

Reply:
in think the env variable is actually INFLUX_TOKEN: https://github.com/influxdata/influxdb/issues/16048#issuecomment-562664102

Reply:
I think I prefer this being explicit and people knowing that we're setting it to the value of an environment variable, rather than it implicitly using it